tss-esapi/src/abstraction/transient/key_attestation.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::{ObjectWrapper, TransientKeyContext};
+use crate::{
+    abstraction::ek,
+    constants::SessionType,
+    handles::{AuthHandle, KeyHandle, SessionHandle},
+    interface_types::{
+        algorithm::{AsymmetricAlgorithm, HashingAlgorithm},
+        session_handles::{AuthSession, PolicySession},
+    },
+    structures::{EncryptedSecret, IDObject, SymmetricDefinition},
+    tss2_esys::{TPM2B_PUBLIC, TPMT_PUBLIC},
+    utils::PublicKey,
+    Result,
+};
+use std::convert::{TryFrom, TryInto};
+
+#[derive(Debug)]
+/// Wrapper for the parameters needed by MakeCredential
+pub struct MakeCredParams {
+    /// TPM name of the object
+    name: Vec<u8>,
+    /// Encoding of the public parameters of the object whose name
+    /// will be included in the credential computations
+    public: Vec<u8>,
+    /// Public part of the key used to protect the credential
+    attesting_key_pub: PublicKey,
+}
+
+impl MakeCredParams {
+    pub fn name(&self) -> &[u8] {
+        &self.name
+    }
+
+    pub fn public(&self) -> &[u8] {
+        &self.public
+    }
+
+    pub fn attesting_key_pub(&self) -> &PublicKey {
+        &self.attesting_key_pub
+    }
+}
+
+impl TransientKeyContext {
+    /// Get the data required to perform a MakeCredential
+    ///
+    /// # Parameters
+    ///
+    /// * `object` - the object whose TPM name will be included in
+    /// the credential
+    /// * `key` - the key to be used to encrypt the secret that wraps
+    /// the credential
+    ///
+    /// **Note**: If no `key` is given, the default Endorsement Key
+    /// will be used.  
+    pub fn get_make_cred_params(
+        &mut self,
+        object: ObjectWrapper,
+        key: Option<ObjectWrapper>,
+    ) -> Result<MakeCredParams> {
+        let object_handle = self.load_key(object.params, object.material, None)?;
+        let (object_public, object_name, _) =
+            self.context.read_public(object_handle).or_else(|e| {
+                self.context.flush_context(object_handle.into())?;
+                Err(e)
+            })?;
+        self.context.flush_context(object_handle.into())?;
+
+        let public = TPM2B_PUBLIC::from(object_public);
+        let public = unsafe {
+            std::mem::transmute::<TPMT_PUBLIC, [u8; std::mem::size_of::<TPMT_PUBLIC>()]>(
+                public.publicArea,
+            )
+        };
+        let attesting_key_pub = match key {
+            None => {
+                let key_handle =
```

Comment:
Would it possible to put this in a ```get_ek_object_public``` method?

Reply:
Done!